tests/func/test_data_cloud.py

```diff
@@ -585,24 +583,21 @@ def test(self):
         self._test_recursive_pull()
 
 
-class TestCheckSumRecalculation(TestDvc):
-    def test(self):
-        test_get_file_checksum = spy(RemoteLOCAL.get_file_checksum)
-        with patch.object(
-            RemoteLOCAL, "get_file_checksum", test_get_file_checksum
-        ):
-            url = Local.get_url()
-            ret = main(["remote", "add", "-d", TEST_REMOTE, url])
-            self.assertEqual(ret, 0)
-            ret = main(["config", "cache.type", "hardlink"])
-            self.assertEqual(ret, 0)
-            ret = main(["add", self.FOO])
-            self.assertEqual(ret, 0)
-            ret = main(["push"])
-            self.assertEqual(ret, 0)
-            ret = main(["run", "-d", self.FOO, "echo foo"])
-            self.assertEqual(ret, 0)
-        self.assertEqual(test_get_file_checksum.mock.call_count, 1)
+def test_checksum_recalculation(mocker, dvc, tmp_dir):
+    tmp_dir.gen({"foo": "foo"})
+    test_get_file_checksum = mocker.spy(RemoteLOCAL, "get_file_checksum")
+    url = Local.get_url()
+    ret = main(["remote", "add", "-d", TEST_REMOTE, url])
```

Comment:
No need to use `TEST_REMOTE`, you can use some string name, like "upstream"

Reply:
@pared I think it is out of scope for this change 🙂